### DISCUSSION TITLE
[Bug]: Pimcore Documents - Images not loading

### DISCUSSION THREAD

**Discussion by dennSted** [pimcore/pimcore#17329] [Q&A / Support]
### Pimcore version

v11.2.3

### Steps to reproduce

Web2Print Settings: Gotenberg Chromium
Document Save Mode: cleanup

I'm trying to create customs specs documents including some product related elements like Sizes, Measurements, Position Images etc. I have added a new Action inside Web2PrintController e.g.

```
    public function specsLabelAction(Request $request) : Response
    {
        return $this->render('web2print/specs_label.html.twig');
    }
```
and the corresponding twig file:
```
{% extends 'web2print/testpagelayout.html.twig' %}

{% block content %}

    {% if editmode %}
        <p>Select product for this PDF Generation:</p>
        {{ pimcore_relation("product") }}
    {% else %}
        {% set pro= pimcore_relation("product").element %} 
        {% if pro %} 
            <span class="headerprod">{{pro.ProdNo}}-{{pro.RevCode}} - {{ pro.ProdName }}</span>
            <h2 class="txt-italic">C. LABEL</h2>
            {# Main/Care Label Section #}
            <div>
                <table class="w-100">
                    <tr>
                        <td><p class="headl">Main/Care Label</p></td>
                        <td><p class="txt-center"><strong>Size Range:</strong> {{ pro.ProSizeRange }}</p></td>                        
                    </tr>
                </table>
                <div class="txt-center">
                    <img src="{{ pro.ProMainCareLabelImage }}" alt="Main/Care Label Image" class="fitted"/>
                </div>
            </div>
            {# Size Label Section #}
            <div>
                <table class="w-100">
                    <tr>
                        <td><p class="headl">{{ pro.ProSizeLabel }}</p></td>
                        <td><p class="txt-right"><strong>Size:</strong> {{ pro.ProSizeSize|nl2br }} </p></td>                        
                    </tr>
                </table>
                <div class="txt-center">
                    <img src="http://pimcore.mydomain.de{{ pro.SizeLabelImage }}" alt="Size Label Image" class="fitted"/>
                </div>
                <p><strong>Position:</strong> <br/>
                    {{ pro.SizeLabelPlacement|nl2br }} 
                </p>
            </div>

        {% else %}
            <p>Product not set</p>

        {% endif %}
    {% endif %}

{% endblock %}
```

I have also tried to use the getThumbnail() method

```
{% set assetimg = asset(pro.SizeLabelImage) %}
{{ assetimg.getThumbnail('print_test').getHtml() | raw }}
```

The result is the same. It's not loading the image/s inside the document - only the default web placeholder with alt text.
<img width="244" alt="image-not-loading-pimcore-document" src="https://github.com/pimcore/pimcore/assets/170010966/0c4f6437-60c7-47ff-8440-833c27eb0df7">

When i try to output only the content of e.g. "pro.SizeLabelImage" i get the following:
`
/Product%20Images/Woven%20Size%20Label/XXS-5XL-01.png
`
That's the reason why I also tried to add the Domain to the img src. When trying out the combination inside the browser, it displayed the corresponding image asset.
`
http://pimcore.mydomain.de/Product%20Images/Woven%20Size%20Label/XXS-5XL-01.png
`

### Actual Behavior

When generating the PDF file all default HTML elements are displayed correctly with the connected Pimcore data (like Size data, texts etc.). But the Images aren't visible

### Expected Behavior

Images should load as it already does it in the HTML preview

**Comment by fashxp** [comment]
hmm, I cannot reproduce this on our demo. 
what happens when you try to load the image directly in the browser ... what response do you get. seems that somehow the image cannot be delivered...

**Reply by dennSted** [reply]
Hi fashxp,

when I try to load the images directly in the browser they are displayed normally. The images are also visible inside the "Preview". Only while generating the PDF they are not converting.

One special thing, I'm not sure if it has something to do with it, is that currently our pimcore instance is running on a server which is only reachable within a certain network...

**Reply by dennSted** [reply]
Small update,

the problem has something to do with a network setting of the docker container where the Gotenberg / PDFreactor is running. In my case it needed to be changed from bridge to host. This solved the image generating issue aswell as the processing time for generating the images.
